### PR TITLE
Implementation

### PR DESCRIPTION
**Pull request by prolic**

### CODE REVIEW

File: src/Api/Translation.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        if (strpos($response->getHeaderLine('Content-Type'), 'application/json') !== 0) {
+            throw new Exception\HydrationException('The ModelHydrator cannot hydrate response with Content-Type:'.$response->getHeaderLine('Content-Type'));
+        }
+
+        $data = json_decode($body, true);
+        if (JSON_ERROR_NONE !== json_last_error()) {
+            throw new Exception\HydrationException(sprintf('Error (%d) when trying to json_decode response', json_last_error()));
+        }
+
+        foreach ($data as $translation) {
```
Comment (prolic) [inline src/Api/Translation.php]: There is no API call to get a single translation from phraseapp, but I can get a list of translations filtered by domain. It the workaround okay like this?

File: src/HttpClientConfigurator.php [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    /**
+     * @var string
+     */
+    private $token;
+
+    /**
+     * @param HttpClient|null $httpClient
+     * @param UriFactory|null $uriFactory
+     */
+    public function __construct(HttpClient $httpClient = null, UriFactory $uriFactory = null, string $token)
```
Comment (prolic) [inline src/HttpClientConfigurator.php]: Token is injected in the contructor, is that correct?

File: src/HttpClientConfigurator.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+    /**
+     * @return HttpClient
+     */
+    public function createConfiguredClient(): HttpClient
+    {
+        $plugins = $this->prependPlugins;
+
+        $plugins[] = new Plugin\AuthenticationPlugin(new BasicAuth($this->token, ''));
```
Comment (prolic) [inline src/HttpClientConfigurator.php:77]: Token is used here.

**Comment by prolic** [comment]
More API calls following soon.

File: src/Api/Translation.php [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+     *
+     * @param string $projectKey
+     * @param string $id
+     * @param string $localeId
+     *
+     * @return TranslationModel|ResponseInterface
+     *
+     * @throws Exception
+     */
+    public function get(string $projectKey, string $domain, string $id, string $localeId)
```
Comment (Nyholm) [inline src/Api/Translation.php]: You do not need domain. I think we should be more general in the client. Maybe use $query as parameter?

File: src/Api/Translation.php [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+namespace FAPI\PhraseApp\Api;
+
+use FAPI\PhraseApp\Model\Translation\TranslationDeleted;
+use FAPI\PhraseApp\Model\Translation\Translation as TranslationModel;
+use GuzzleHttp\Psr7\Response;
+use Psr\Http\Message\ResponseInterface;
+use FAPI\PhraseApp\Exception;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
```
Comment (Nyholm) [inline src/Api/Translation.php]: You should add your name. The names here is the developers "who knows most about this class"

File: src/Api/Translation.php [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        if (!$this->hydrator) {
+            return $response;
+        }
+
+        if ($response->getStatusCode() !== 200) {
+            $this->handleErrors($response);
+        }
+
+        $body = $response->getBody()->__toString();
+        if (strpos($response->getHeaderLine('Content-Type'), 'application/json') !== 0) {
```
Comment (Nyholm) [inline src/Api/Translation.php]: Isn't the API always returning json?

Anyhow, you cannot be sure this->hydrator is a ModelHydrator. This test should be removed.

File: src/Api/Translation.php (hunk 1 above)
Comment (Nyholm) [reply src/Api/Translation.php]: No, the API client should reflect the API server. If the server does not support fetching a single translation, the client shouldn't do that either. 

It sucks, yes, but we can only blame the API server.

File: src/HttpClientConfigurator.php (hunk 2 above)
Comment (Nyholm) [reply src/HttpClientConfigurator.php]: Sure, but put it before the optional parameters. Also make sure to update the doc block

File: src/HttpClientConfigurator.php (hunk 3 above)
Comment (Nyholm) [reply src/HttpClientConfigurator.php:77]: Is BasicAuth right?

**Review by Nyholm — CHANGES_REQUESTED** [review]
Good. Most thinks looks great. I had some minor comments.

File: src/HttpClientConfigurator.php (hunk 3 above)
Comment (prolic) [reply src/HttpClientConfigurator.php:77]: yes

Comment (Nyholm) [reply src/HttpClientConfigurator.php:77]: Reading: https://phraseapp.com/docs/api/v2/#authentication

You are correct, but they are doing APIs wrong =)

**Comment by prolic** [comment]
I will only add import and export api, the rest is not available without worksaround and multiple api calls. Import is already done. Export will follow later that day.

**Comment by Nyholm** [comment]
Yes, sure. Only add the functions that you need. The rest can be added later. 

Remember that the API client should reflect methods on the API server. Not common use cases. So multiple API calls for one function should never be needed.

**Comment by prolic** [comment]
@Nyholm Ready for review

**Review by Nyholm — APPROVED** [review]

**Comment by Nyholm** [comment]
Awesome. I'll merge this now and register styleCI and packagist etc.

**Comment by Nyholm** [comment]
https://packagist.org/packages/friendsofapi/phraseapp

Thank you @prolic

**Pull request merged**